test/lit/passes/vacuum-eh.wast

```diff
@@ -0,0 +1,254 @@
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (i32.const 2)
+  ;; CHECK-NEXT: )
+  (func $inner-try_table-catch_all-test (result i32)
+    (local $0 i32)
+    ;; The exception thrown in the inner try_table is caught by the inner
+    ;; catch_all, so the outer try_table body does not throw and the outer
+    ;; try_table can be removed
+    (block $outer-tryend
+      (drop
+        (block $outer-catch (result i32)
+          (try_table (catch $e $outer-catch)
+            (block $inner-catch
+              (try_table (catch_all $inner-catch)
+                (throw $e (i32.const 0))
+              )
+              (unreachable)
+            )
+            (return (i32.const 1))
+          )
+          (br $outer-tryend)
+        )
+      )
+    )
+    (i32.const 2)
+  )
+
+  ;; CHECK:      (func $inner-try_table-catch-test (type $void)
+  ;; CHECK-NEXT:  (local $0 i32)
+  ;; CHECK-NEXT:  (block $outer-tryend
+  ;; CHECK-NEXT:   (drop
+  ;; CHECK-NEXT:    (block $outer-catch (result i32)
+  ;; CHECK-NEXT:     (try_table (catch $e $outer-catch)
+  ;; CHECK-NEXT:      (block $inner-tryend
+  ;; CHECK-NEXT:       (drop
+  ;; CHECK-NEXT:        (block $inner-catch (result i32)
+  ;; CHECK-NEXT:         (try_table (catch $e $inner-catch)
+  ;; CHECK-NEXT:          (throw $e2
+  ;; CHECK-NEXT:           (i32.const 0)
+  ;; CHECK-NEXT:          )
+  ;; CHECK-NEXT:         )
+  ;; CHECK-NEXT:        )
+  ;; CHECK-NEXT:       )
+  ;; CHECK-NEXT:       (local.set $0
+  ;; CHECK-NEXT:        (i32.const 1)
+  ;; CHECK-NEXT:       )
+  ;; CHECK-NEXT:      )
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:     (br $outer-tryend)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $inner-try_table-catch-test (local $0 i32)
+    ;; The exception thrown in the inner try_table will not be caught by the
+    ;; inner catch, so the outer try_table cannot be removed.
+    (block $outer-tryend
+      (drop
+        (block $outer-catch (result i32)
+          (try_table (catch $e $outer-catch)
+            (block $inner-tryend
+              (drop
+                (block $inner-catch (result i32)
+                  (try_table (catch $e $inner-catch)
+                    (throw $e2 (i32.const 0))
+                  )
+                  (br $inner-tryend)
+                )
+              )
+              (local.set $0 (i32.const 1))
+            )
+          )
+          (br $outer-tryend)
+        )
+      )
+    )
+  )
+
+  ;; CHECK:      (func $br-in-catch (type $void)
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (unreachable)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $br-in-catch
+    ;; When the try_table is removed, the removal of the implicit branch to the
+    ;; catch target should be registered so that its type will be correctly
+    ;; updated to unreachable.
+    (drop
+      (block $catch (result i32)
+        (try_table (catch $e $catch)
+          (unreachable)
+        )
+        (unreachable)
+      )
+    )
+  )
+
```

Comment:
```suggestion
```

The original test has a `br` within the `catch` body (hence the name) but this doesn't. And that test was added in #2765, which fixed a bug that `typeUpdater.noteRecursiveRemoval(curr->catchBody);` was missing for the catch body. Given that we don't have 'catch bodies' anymore and don't need that `noteRecursiveRemoval`, I think this test can be removed.